src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -57,8 +57,6 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     val blockchain = BlockchainImpl(storages)
 
     val blockExecResult = for {
-      //todo why do we require in executeBlockTransactions separate blockchain and blockchain storage
-      //todo blockchain storage is part of blockchain
```

Comment:
As we are currently using `Blockchain` to get state related data, what about if blockchain also allows to retrieve the WorldStateProxy, something like `def getWorldStateProxy(stateRootHash: Option[ByteString] = None): WS` and let BlockchainImple to create the WorldStateProxy using the storages it has.

I have a working version [here](https://github.com/input-output-hk/etc-client/commit/fdcb131512ac3b66871acad6e259363739662f00#diff-abb3f698c5418831dcb5f183fbf61937R243) as I needed to do some work in that code (If you think it's not ok, I can do a rollback tho).